src/ert/gui/model/real_list.py

```diff
@@ -107,91 +90,71 @@ def parent(self, child: Optional[QModelIndex] = None) -> Optional[QObject]:
     def index(
         self, row: int, column: int, parent: Optional[QModelIndex] = None
     ) -> QModelIndex:
-        if parent is None:
-            parent = QModelIndex()
-        if parent.isValid():
-            return QModelIndex()
-        real_index = self.mapToSource(self.createIndex(row, 0, parent))
-        ret_index = self.createIndex(row, column, real_index.data(NodeRole))
-        return ret_index
+        parent = parent if parent else QModelIndex()
+        if not parent.isValid():
+            real_index = self.mapToSource(self.createIndex(row, 0, parent))
+            return self.createIndex(row, column, real_index.data(NodeRole))
+        return QModelIndex()
 
     @override
     def hasChildren(self, parent: Optional[QModelIndex] = None) -> bool:
         # Reimplemented, since in the source model, the realizations have
         # children (i.e. valid indices.). Realizations do not have children in
         # this model.
-        if parent is None:
-            parent = QModelIndex()
-        if parent.isValid():
-            return False
-        source_model = self.sourceModel()
-        assert source_model is not None
-        return source_model.hasChildren(self.mapToSource(parent))
+        parent = parent if parent else QModelIndex()
+        if not parent.isValid():
+            source_model = self.sourceModel()
+            assert source_model is not None
+            return source_model.hasChildren(self.mapToSource(parent))
+        return False
 
     def mapToSource(self, proxyIndex: QModelIndex) -> QModelIndex:
-        if not proxyIndex.isValid():
-            return QModelIndex()
-        sm = self.sourceModel()
-        assert sm is not None
-        iter_index = sm.index(self._iter, 0, QModelIndex())
-        if not iter_index.isValid() or not sm.hasChildren(iter_index):
-            return QModelIndex()
-        real_index = sm.index(proxyIndex.row(), proxyIndex.column(), iter_index)
-        return real_index
-
-    def mapFromSource(self, sourceIndex: QModelIndex) -> QModelIndex:
-        if not sourceIndex.isValid():
-            return QModelIndex()
-        source_node = sourceIndex.internalPointer()
-        if source_node is None:
-            return QModelIndex()
-        if not self._accept_index(sourceIndex):
-            return QModelIndex()
-        return self.index(sourceIndex.row(), sourceIndex.column(), QModelIndex())
+        if proxyIndex.isValid():
+            sm = self.sourceModel()
+            assert sm is not None
+            iter_index = sm.index(self._iter, 0, QModelIndex())
+            if iter_index.isValid() and sm.hasChildren(iter_index):
+                return sm.index(proxyIndex.row(), proxyIndex.column(), iter_index)
+        return QModelIndex()
+
+    def mapFromSource(self, src_index: QModelIndex) -> QModelIndex:
+        if src_index.isValid():
+            source_node = src_index.internalPointer()
+
+            if source_node and self._accept_index(src_index):
```

Comment:
could drop the source_node temp variable. it isn't used after this test

Reply:
Agreed. That made the function more or less identical to the job_list counterpart too. 👍